cmake/Toolchain-gcc-clang.cmake

```diff
@@ -2,12 +2,14 @@ if(X86_32 OR X86_64)
     add_compile_options(-mfpmath=sse -msse2)
 endif()
 
-if(X86_64)
-    # Require and optimize for Core2 level support, tune for generic.
-    add_compile_options(-march=core2 -mtune=generic)
-elseif(X86_32)
-    # Optimize for pentium-mmx and tune for generic for older builds.
-    add_compile_options(-march=pentium-mmx -mtune=generic)
+if(UPSTREAM_RELEASE)
```

Comment:
I'm not opposed to the change, but why limit these options to `UPSTREAM_RELEASE`?